### PR TITLE
Synced children related letters

### PR DESCRIPTION
**Pull request by SokyranTheDragon**

### CODE REVIEW

File: Source/Client/Patches/Letters.cs [hunk 1]
```diff
@@ -31,17 +31,31 @@ static void Postfix(LetterStack __instance)
...
             {
-                method.Invoke(__instance);
-                return false;
+                if (rejectMethods.TryGetValue(__instance.GetType(), out var method))
+                {
+                    method.Invoke(__instance);
+                    return false;
+                }
+                // Special case - no predefined reject option. Keep the current state (as either colonist or slave)
+                if (__instance is ChoiceLetter_BabyToChild babyToChild)
```
Comment (Zetrith) [inline Source/Client/Patches/Letters.cs:50]: A more elegant way would be to package the special handler as a delegate in the rejectMethods map. And maybe rename the map to defaultChoices or something.

Comment (SokyranTheDragon) [reply Source/Client/Patches/Letters.cs]: I've modified how it works. I'm not the happiest about how it looks, but it works.

**Comment by SokyranTheDragon** [comment]
I've included baby naming dialog in this PR, as it plays nicely with the changes to choice letters

**Comment by SokyranTheDragon** [comment]
Also synced the growth moment letter

File: Source/Client/Patches/Letters.cs (hunk 1 above)
Comment (Zetrith) [reply Source/Client/Patches/Letters.cs]: The name RegisterMethod is too generic and I think this breaks ChoiceLetter_RansomDemand.

**Pull request merged**